StripePayments/StripePayments/Source/Captcha/PassiveCaptcha.swift

```diff
@@ -0,0 +1,138 @@
+        self.passiveCaptcha = passiveCaptcha
+        self.testTimeout = testTimeout
+    }
+
+    @_spi(STP) public func start() {
+        guard let passiveCaptcha, validationTask == nil else { return }
+        let timeoutNs: UInt64 = {
+            if let testTimeout {
+                return testTimeout
+            }
+            return STPAnalyticsClient.isUnitOrUITest ? 0 : 6_000_000_000
+        }()
+
+        validationTask = Task<String?, Never> { [siteKey = passiveCaptcha.siteKey, rqdata = passiveCaptcha.rqdata, timeoutNs] () -> String? in
+            do {
+                let hcaptcha = try HCaptcha(apiKey: siteKey,
+                                            passiveApiKey: true,
+                                            rqdata: rqdata,
+                                            host: "stripecdn.com")
+                STPAnalyticsClient.sharedClient.logPassiveCaptchaInit(siteKey: siteKey)
+
+                return await withCheckedContinuation { (continuation: CheckedContinuation<String?, Never>) in
+                    var hasCompleted = false
+                    let timeoutTask = Task { @MainActor in
+                        try? await Task.sleep(nanoseconds: timeoutNs)
+                        if hasCompleted { return }
+                        hasCompleted = true
+                        STPAnalyticsClient.sharedClient.logPassiveCaptchaTimeout(siteKey: siteKey)
+                        continuation.resume(returning: nil)
+                    }
+
+                    hcaptcha.didFinishLoading {
+                        hcaptcha.validate { result in
+                            Task { @MainActor in
+                                if hasCompleted { return }
+                                do {
+                                    let token = try result.dematerialize()
+                                    hasCompleted = true
+                                    STPAnalyticsClient.sharedClient.logPassiveCaptchaSuccess(siteKey: siteKey)
+                                    timeoutTask.cancel()
+                                    continuation.resume(returning: token)
+                                } catch {
+                                    hasCompleted = true
+                                    STPAnalyticsClient.sharedClient.logPassiveCaptchaError(error: error, siteKey: siteKey)
+                                    timeoutTask.cancel()
+                                    continuation.resume(returning: nil)
+                                }
+                            }
+                        }
+                    }
+                }
+            } catch {
+                STPAnalyticsClient.sharedClient.logPassiveCaptchaError(error: error, siteKey: siteKey)
+                return nil
+            }
+        }
+    }
+
+    @_spi(STP) public func fetchToken() async -> String? {
+        if validationTask == nil {
+            start()
```

Comment:
Could we have an analytic here too? That way we could be alerted if this is slowing down the payment. I'd hope that most of the time, the token would already be ready by the time we call `fetchToken()`